lua/lspconfig.lua

```diff
@@ -7,41 +7,46 @@ local M = {
 ---@class Alias
 ---@field to string The new name of the server
 ---@field version string The version that the alias will be removed in
----@param name string
----@return Alias
-local function server_alias(name)
-  local aliases = {
-    ['fennel-ls'] = {
-      to = 'fennel_ls',
-      version = '0.2.0',
-    },
-    ruby_ls = {
-      to = 'ruby_lsp',
-      version = '0.2.0',
-    },
-    ['starlark-rust'] = {
-      to = 'starlark_rust',
-      version = '0.2.0',
-    },
-    sumneko_lua = {
-      to = 'lua_ls',
-      version = '0.2.0',
-    },
-    tsserver = {
-      to = 'ts_ls',
-      version = '0.2.0',
-    },
-  }
+---@field inconfig? boolean need shown in lspinfo
+local aliases = {
+  ['fennel-ls'] = {
+    to = 'fennel_ls',
+    version = '0.2.1',
+  },
+  ruby_ls = {
+    to = 'ruby_lsp',
+    version = '0.2.1',
+  },
+  ['starlark-rust'] = {
+    to = 'starlark_rust',
+    version = '0.2.1',
+  },
+  sumneko_lua = {
+    to = 'lua_ls',
+    version = '0.2.1',
+  },
+  tsserver = {
+    to = 'ts_ls',
+    version = '0.2.1',
+  },
+}
 
-  return aliases[name]
-end
+M.server_aliases = setmetatable(aliases, {
+  __index = function(t, k)
+    return rawget(t, k) or nil
+  end,
+  __newindex = function()
+    vim.notify('Attempt to modify a frozen table aliases', vim.log.levels.ERROR)
```

Comment:
why are we doing this instead of just exposing it as a function?

Reply:
also thinking about this. one is closure one is freeze .  prefer use a closure ?

Reply:
freeze is uncommon, we should use plain old simple patterns except when we have a strong need to do something more clever.

Reply:
Knowledge